src/include/duckdb/main/client_context.hpp

```diff
@@ -42,6 +42,8 @@ class ClientContext : public std::enable_shared_from_this<ClientContext> {
 	DUCKDB_API explicit ClientContext(shared_ptr<DatabaseInstance> db);
 	DUCKDB_API ~ClientContext();
 
+	//! Previous Query profiler
+	QueryProfiler prev_profiler;
```

Comment:
What exactly is the reason for this prev_profiler? Can't you just read `profiler` instead?

Reply:
The profiler, always refer to the current query which in this case would be PRAGMA last_profiling_output, 
PRAGMA detailed_profiling_output or PRAGMA all_profiling_output, while we need the profiler information from the prev query.

Reply:
Then this solution will only work for the first `PRAGMA` that is fired. If you fire e.g. `PRAGMA last_profiling_output` twice, the second call will not work anymore. There is already machinery in the profiler to decide whether or not a query requires profiling (see `OperatorRequiresProfiling`). Can't that be used so that calls to e.g. `PRAGMA last_profiling_output` will not overwriting the last output profiling?